### PR TITLE
Adding FactUshauriSMSReminders

### PR DESCRIPTION
**Pull request by Marymary-dev**

### CODE REVIEW

File: Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql [hunk 1]
```diff
@@ -0,0 +1,118 @@
+IF OBJECT_ID(N'[NDWH].[dbo].[FACTUshauriSMSReminders]', N'U') IS NOT NULL DROP TABLE [NDWH].[dbo].[FACTUshauriSMSReminders];
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql:1]: @Marymary-dev  nitpik: worth maintaining the naming convention of the tables with CamelCase;
`FactUshauriSmsReminders`

File: Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+   Select
+      FactKey = IDENTITY(INT, 1, 1),
+      facility.FacilityKey,
+      partner.PartnerKey,
+      patient.PatientKey,
+      agency.AgencyKey,
+      EOMONTH(Try_CONVERT(date, AppointmentDate)) AS AsofDate,
+      apt.MaritalStatus,
+      age_group.AgeGroupKey,
+      cast (appointment.Datekey as date) as AppointmentDate,
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql:21]: @Marymary-dev for the dates in the fact, we need to maintain them as date keys that can be joined later  in the reporting layer.

we can also have all the date keys with the suffix  `...DateKey` for consistency e.g. `AppointmentDateKey`

File: Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      MissedAppointmentSMSDeliveryStatus,
+      MissedAppointmentSMSDeliveryFailureReason,
+      TracingCalls,
+      TracingSMS,
+      TracingHomeVisits,
+      TracingOutcome,
+      Cast (TracingDate.Datekey as date) As TracingOutcomeDate,
+      Cast (DateReturnedToCare.DateKey as date ) as DateReturnedToCare,
+      DaysDefaulted,
+      NUPIHash into NDWH.dbo.FACTUshauriSMSReminders
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql:63]: nitpik: newline after NupiHash

File: Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+         on OneDayDate.Date = Try_Convert(date, apt.OneDaySMSSendDate) 
+      left join
+         NDWH.dbo.DimDate as MissedAppointmentDate 
+         on MissedAppointmentDate.Date = Try_Convert(date, apt.MissedAppointmentSMSSendDate) 
+      left join
+         NDWH.dbo.DimDate as TracingDate 
+         on TracingDate.Date = Try_Convert(date, apt.TracingOutcomeDate) 
+      left join
+         NDWH.dbo.DimDate as DateReturnedToCare 
+         on DateReturnedToCare.Date = Try_Convert(date, apt.DateReturnedToCare) alter table NDWH.dbo.FACTUshauriSMSReminders add primary key(FactKey);
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql:117]: nitpik: newline after `Try_Convert(date, apt.DateReturnedToCare)`

File: Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                             Sdp,
+                             Sdp_agency AS Agency
+             FROM   Ods.Dbo.All_emrsites)
+
+    SELECT FactKey = IDENTITY(Int, 1, 1),
+           Facility.Facilitykey,
+           Partner.Partnerkey,
+           Patient.Patientkey,
+           Agency.Agencykey,
+           EOMONTH(TRY_CONVERT(date, Appointmentdate)) AS AsofDate, ---This date has conversion issues----
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql:16]: @Marymary-dev this should also be a datekey ?

Comment (Marymary-dev) [reply Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql:16]: @nobert-mumo  there is a comment besides it

File: Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+                             Sdp,
+                             Sdp_agency AS Agency
+             FROM   Ods.Dbo.All_emrsites)
+
+    SELECT FactKey = IDENTITY(Int, 1, 1),
+           Facility.Facilitykey,
+           Partner.Partnerkey,
+           Patient.Patientkey,
+           Agency.Agencykey,
+           Apt.Maritalstatus,
```
Comment (nobert-mumo) [inline Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql:16]: @Marymary-dev the `Maritalstatus` can come from DimPatient ?

Comment (Marymary-dev) [reply Scripts/NDWH/C&T FACT TABLES/load_FactUshauriSMSReminders.sql]: @nobert-mumo  sure, thanks,

**Review by nobert-mumo — APPROVED** [review]
LGTM!

**Pull request merged**